# Config files

Here are some of my configuration files for working R and JHPCE.

## R setup

* RStudio Desktop _Preview_ version: https://rstudio.com/products/rstudio/download/preview/
* R: latest versions that match Bioconductor release and devel
  - macOS: you can download the latest "branch" and "devel" from https://mac.r-project.org/
  - Windows: https://cran.r-project.org/, also needs Rtools40 https://cran.r-project.org/bin/windows/Rtools/
* RSwitch
  - macOS: https://rud.is/rswitch/
  - Windows: I install the R versions in two different folders and switch them using RStudio's global configuration options
 https://cran.r-project.org/
* Bioconductor-devel docker image: http://bioconductor.org/help/docker/

Additional software:

- Both: [Notion](https://www.notion.so/desktop), [Slack](https://slack.com/downloads)
- Windows: [WinSCP](https://sourceforge.net/projects/winscp/), [Git bash](https://gitforwindows.org/), [PuTTY](https://www.putty.org/), [Notepad++](https://notepad-plus-plus.org/downloads/), [NppToR](https://sourceforge.net/projects/npptor/)
- macOS: [Cyberduck](https://cyberduck.io/), [iTerm2](https://www.iterm2.com/), [Alfred](https://www.alfredapp.com/), [TextMate2](https://macromates.com/download)

You might also be interested in checking out our older [onboarding setup materials](http://research.libd.org/jaffe_onboarding/setup.html).

### R packages

The following R code installs all the packages I have installed currently as my _base_. As I explore packages, I might try new ones that are not in this list.

```{r "install_from_scratch", eval = FALSE}
## Install from scratch
if (!requireNamespace("remotes", quietly = TRUE))
    install.packages("remotes")
remotes::install_cran("BiocManager")
BiocManager::version()

## Rprofile packages
remotes::install_github(c(
    "jalvesaq/colorout"
))
remotes::install_cran(c(
    "devtools",
    "usethis"
))

## Temporary for now due to https://github.com/ropensci/RefManageR/issues/79
remotes::install_github(c(
    "ropensci/bibtex",
    "ropensci/RefManageR",
    "cboettig/knitcitations"
))

## Main packages
BiocManager::install(c(
    "biocthis",
    "brainflowprobes",
    "derfinder",
    "derfinderPlot",
    "GenomicState",
    "megadepth",
    "qsvaR",
    "recount",
    "recountWorkflow",
    "recount3",
    "regutools",
    "regionReport",
    "spatialLIBD",
    "TREG"
), dependencies = TRUE, update = FALSE)

## LIBD packages
remotes::install_github(c(
    "LieberInstitute/jaffelab",
    "LieberInstitute/recount.bwtool",
    "LieberInstitute/sgejobs",
    "LieberInstitute/shinycsv"
), dependencies = TRUE)

## CRAN packages
remotes::install_cran(c(
    "ari",
    "bookdown",
    "blogdown",
    "clue",
    "corrplot",
    "getopt",
    "ggpubr",
    "ggrepel",
    "ggthemes",
    "googlesheets4",
    "MatrixEQTL",
    "pagedown",
    "patchwork",
    "postcards",
    "reprex",
    "rsconnect",
    "rtweet",
    "suncalc",
    "tidyverse",
    "xaringan",
    "xaringanthemer",
    "UpSetR",
    "VennDiagram"
))

## Bioc packages
BiocManager::install(c(
    "BiocCheck",
    "ComplexHeatmap",
    "ExploreModelMatrix",
    "iSEE",
    "scater",
    "scran",
    "sva",
    "variancePartition"
), update = FALSE)

## GitHub packages
remotes::install_github(c(
    "cloudyr/rmote",
    "gadenbuie/rsthemes",
    "gadenbuie/xaringanExtra"
))

## Private packages (you likely won't have access to them)
remotes::install_github(c(
    "LieberInstitute/LIBDpheno",
    "LieberInstitute/libdRSE"
), dependencies = TRUE)
```

#### Previously installed

As I described in [my blog post on updating R](http://lcolladotor.github.io/2017/05/04/updating-r/#.X6ICiIhKguU), I also like to saved my list of currently installed packages.

```{r "previously_installed", eval = FALSE}
installed_path <- "/Users/lcollado/Dropbox/Computing/R"
installed <- dir(.libPaths())
save(installed, file = file.path(installed_path, paste0(Sys.Date(), '-installed.Rdata')))
``` 

```{r "currently_installed", eval = FALSE}
## Load previous "installed" packaages
previous <- dir(path = installed_path, pattern = 'installed.Rdata')
load(file.path(installed_path, previous[length(previous)]), verbose = TRUE)

## Locate current pacakges
current <- dir(.libPaths())
## Missing
installed[!installed %in% current]

## Install missing packages
BiocManager::install(installed[!installed %in% current])

## New
current[!current %in% installed]
```


## R config files

### `~/.Rprofile`

Edit it with `usethis::edit_r_profile()`. Don't add any analysis packages here as discussed in the "what they forgot to teach you about R" workshop.

```{r "r_profile", eval = FALSE}
## Change colors
# Source https://github.com/jalvesaq/colorout
if(Sys.getenv('TERM') %in% c("term", "xterm-256color", "cygwin", "screen")) {
    if (!requireNamespace("colorout", quietly = TRUE) & .Platform$OS.type != 'windows') {
        cat('To install colorout use: remotes::install_github("jalvesaq/colorout")\n')
    }
}

# https://bookdown.org/yihui/blogdown/global-options.html
options(blogdown.author = 'L. Collado-Torres')
options(blogdown.ext = '.Rmd')
options(blogdown.insertimage.usebaseurl = TRUE)

## From RStudio::conf 2019 Building Tidy Tools
if (interactive()) {
    suppressMessages(require("devtools"))
    suppressMessages(require("usethis"))
    suppressMessages(require("testthat"))
    
    options(
        warnPartialMatchArgs = TRUE,
        warnPartialMatchDollar = TRUE,
        warnPartialMatchAttr = TRUE
    )
}

## https://blog.rstudio.com/2013/06/10/rstudio-cran-mirror/
options(repos = c(CRAN = "https://cloud.r-project.org/"))

## For usethis::use_git()
options(usethis.protocol = "ssh")

## For usethis
options(
    usethis.full_name = "Leonardo Collado-Torres",
    usethis.description = list(
        `Authors@R` = 'c(
    person("Leonardo", "Collado-Torres", role = c("aut", "cre"),
    email = "lcolladotor@gmail.com", comment = c(ORCID = "0000-0003-2140-308X"))
    )'
    )
)

## For biocthis
options("biocthis.pkgdown" = TRUE)
options("biocthis.testthat" = TRUE)

## From https://www.garrickadenbuie.com/project/rsthemes/
if (interactive() && requireNamespace("rsthemes", quietly = TRUE)) {
    # Set preferred themes if not handled elsewhere..
    rsthemes::set_theme_light("base16 Monokai {rsthemes}")  # light theme
    rsthemes::set_theme_dark("base16 Pop {rsthemes}") # dark theme
    rsthemes::set_theme_favorite(c(
        "base16 Monokai {rsthemes}",
        "base16 Pop {rsthemes}",
        "base16 Brewer {rsthemes}",
        "One Dark {rsthemes}",
        "Solarized Light {rsthemes}"
    ))

    # Whenever the R session restarts inside RStudio...
    setHook("rstudio.sessionInit", function(isNewSession) {
        # Automatically choose the correct theme based on time of day
        ## Used rsthemes::geolocate() once
        rsthemes::use_theme_auto(lat = 39.2891, lon = -76.5583)
    }, action = "append")
}

## From https://twitter.com/hadleywickham/status/1113542388033699840
if(interactive()) {
    if (!requireNamespace("prompt", quietly = TRUE)) {
        cat('To install prompt use: remotes::install_github("gaborcsardi/prompt")\n')
    } else {
        prompt::set_prompt(prompt::prompt_git)
    }
}
```

### `~/.Renviron`

Edit it with `usethis::edit_r_environ()`. Here you can keep personal access tokens (PATs).

```{r "r_environ", eval = FALSE}
GITHUB_PAT=something_provite
TWITTER_PAT=a_private_file_path
```

### `~/.R/Makevars`

I don't have anything in this file since R 4.0 on my macOS laptop.

## Git config files


### `~/.gitconfig`

Edit it with `usethis::edit_git_config()`. Note that my editor is TextMate2 on macOS which can be accessed through the `mate -w` command.

```
[user]
    name = Leonardo Collado Torres
    email = lcolladotor@gmail.com
[core]
    excludesfile = /Users/lcollado/.gitignore_global
    excludesfiles = /Users/lcollado/.gitignore_global
    editor = mate -w
	ignorecase = true
[difftool "sourcetree"]
	cmd = opendiff \"$LOCAL\" \"$REMOTE\"
	path = 
[mergetool "sourcetree"]
	cmd = /Applications/SourceTree.app/Contents/Resources/opendiff-w.sh \"$LOCAL\" \"$REMOTE\" -ancestor \"$BASE\" -merge \"$MERGED\"
	trustExitCode = true
[alias]
    ci = commit
    st = status
    br = branch
    co = checkout
    last = log -1 HEAD
    visual = gitk
[push]
    default = simple
[svn]
    rmdir = true
[commit]
```

### `~/.gitignore_global`

Edit it with `usethis::edit_git_ignore()`. These files are ignored by default by git on my computer.

```
*~

## More at
## https://help.github.com/articles/ignoring-files

# R history #
#############
.Rapp.history
.Rhistory

# LaTeX stuff #
###############
*.aux
*.out


# Compiled source #
###################
*.com
*.class
*.dll
*.exe
*.o
*.so

# Packages #
############
# it's better to unpack these files and commit the raw source
# git has its own built in compression methods
*.7z
*.dmg
*.gz
*.iso
*.jar
*.rar
*.tar
*.zip

# Logs and databases #
######################
*.log
*.sql
*.sqlite

# OS generated files #
######################
.DS_Store
.DS_Store?
._*
.Spotlight-V100
.Trashes
Icon?
ehthumbs.db
Thumbs.db
```

## JHPCE files

### `~/.sge_request`

This file configures the default SGE requests under `qsub` and `qrsh`.

```{bash, eval = FALSE}
# Check http://www.biostat.jhsph.edu/bit/cluster-usage.html for more instructions
#
# Set defaults for mem_free and h_vmem
-l mem_free=12G,h_vmem=12G
#
# Set the standard value for stack size limit
# (needed for some programs to run properly when h_vmem is set)
-l h_stack=256M
#
# Set a default maximum file size that an SGE job can create
-l h_fsize=100G
# Define my email
-M YOUR_EMAIL@gmail.com
# To get an email on a job use -m e
```


### `~/.bashrc`

This is the main configuration file for JHPCE, but it's really for any linux system. Some of the commands have to change when working on macOS.

```{bash, eval = FALSE}
# .bashrc

# Source global definitions
if [ -f /etc/bashrc ]; then
	. /etc/bashrc
fi

# User specific aliases and functions
# Auto-complete command from history
# http://lindesk.com/2009/04/customize-terminal-configuration-setting-bash-cli-power-user/
export INPUTRC=~/.inputrc
# http://www.biostat.jhsph.edu/~afisher/ComputingClub/webfiles/KasperHansenPres/IntermediateUnix.p
df
# https://unix.stackexchange.com/questions/48713/how-can-i-remove-duplicates-in-my-bash-history-pr
eserving-order
export HISTCONTROL=ignoreboth:erasedups
export HISTSIZE=20000
shopt -s histappend
shopt -s cmdhist

# http://superuser.com/questions/384769/alias-rm-rm-i-considered-harmful
alias rmi='rm -i'

# Change command prompt
# http://www.cyberciti.biz/tips/howto-linux-unix-bash-shell-setup-prompt.html
# http://www.cyberciti.biz/faq/bash-shell-change-the-color-of-my-shell-prompt-under-linux-or-unix/
# https://bbs.archlinux.org/viewtopic.php?id=48910
# previous in enigma2: "[\u@\h \W]\$ "
# previously in mac: "\h:\W \u\$ "
export PS1="\[\e[0;33m\]\A \W \$ \[\e[m\]"

# LIDB projects
alias labold="cd /dcl01/lieber/ajaffe/lab"
alias lab="cd /dcl01/ajaffe/data/lab"

## 10X genomics
alias single="cd /dcl01/ajaffe/data/lab/singleCell"
alias spatialold="cd /dcl02/lieber/ajaffe/SpatialTranscriptomics/HumanPilot"
alias spatial="cd /dcl02/lieber/ajaffe/SpatialTranscriptomics/LIBD"
alias matt="cd /dcl01/lieber/ajaffe/Matt/MNT_thesis/snRNAseq/10x_pilot_FINAL"

## Team home directories
alias dcl01="cd /dcl01/lieber/lcolladotor"
alias dcl02="cd /dcl02/lieber/lcolladotor"
alias dcs04="cd /dcs04/lieber/lcolladotor"

## Creating modules
# https://lmod.readthedocs.io/en/latest/050_lua_modulefiles.html
alias modsrc="cd /jhpce/shared/jhpce/libd"
alias modlua="cd /jhpce/shared/jhpce/modulefiles/libd"

# colors
# http://norbauer.com/notebooks/code/notes/ls-colors-and-terminal-app
# used BSD pattern ExGxFxDxBxEgEdxbxgxhxd on http://geoff.greer.fm/lscolors/
# that tool does not specify the colors, which I did by looking manually at
# http://blog.twistedcode.org/2008/04/lscolors-explained.html
# and the norbauer.com site previously mentioned
alias ls="ls --color=auto"
#export LS_COLORS="di=1;34;40:ln=1;36;40:so=1;35;40:pi=1;93;40:ex=1;31;40:bd=1;34;46:cd=1;34;43:su
=0;41:sg=0;46:tw=0;47:ow=0;43"
## After switching to RStudio:
# https://askubuntu.com/questions/466198/how-do-i-change-the-color-for-directories-with-ls-in-the-
console
export LS_COLORS="di=0;32:ln=0;36:so=0;35:pi=0;93:ex=0;31:bd=0;34;46:cd=0;34;43:su=0;41:sg=0;46:tw
=0;47:ow=0;43:fi=0;33"

# Uncomment below for Mac and comment the two previous commands
#export CLICOLOR=1
#export LSCOLORS="ExGxFxDxBxEgEdxbxgxhxd"

## For setup:
# http://erniemiller.org/2011/12/12/textmate-2-rmate-awesome/
## For laptop config:
# http://jonsimpson.co.uk/log/2011/rmate-ssh-remoteforward
## rmate port
# https://github.com/textmate/rmate
export RMATE_PORT="SOME_PORT_YOU_CHOOSE"

## Load the git module by default when qrsh/qsub
## thanks to Jiong Yang
if [[ $HOSTNAME == compute-* ]] || [[ $HOSTNAME == transfer-* ]]; then
    echo "Adding LIBD modules"
    module use /jhpce/shared/jhpce/modulefiles/libd
    echo "Loading git"
    module load git
    module load git-status-size/github
    module load git-lfs/2.8.0
    module load rmate/1.5.9
    module load conda_R/4.1
fi

## To deal with running nextflow without requesting much more memory
## https://jhpce.jhu.edu/question/why-do-i-get-memory-errors-when-running-java/
export _JAVA_OPTIONS="-Xms5g -Xmx6g"

## From https://twitter.com/fellgernon/status/1258455434073124865?s=20
## and https://www.cyberciti.biz/tips/understanding-linux-unix-umask-value-usage.html
umask u=rwx,g=rwx,o= ## equivalent to umask 007
```

### `~/.bash_profile`

This file is basically empty, since all the information is contained in the `~/.bashrc`. That was the recommended setup a few years ago when I asked JHPCE's admins.

```{bash, eval = FALSE}
# .bash_profile

# Get the aliases and functions
if [ -f ~/.bashrc ]; then
	. ~/.bashrc
fi
```

### `~/.inputrc`

Useful for altering the behavior of the up and down arrow keys. See the companion lines on the `~/.bashrc` file.

```{bash, eval = FALSE}
#Page up/page down
"\e[B": history-search-forward
"\e[A": history-search-backward

$include /etc/inputrc
```



